Sources/SwiftDocC/Checker/Checkers/SeeAlsoInTopicsHeadingChecker.swift

```diff
@@ -0,0 +1,56 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import Markdown
+
+public struct SeeAlsoInTopicsHeadingChecker: Checker {
+    public var seeAlsoInTopicsHeadings: [Heading] = []
+    public var problems: [Problem] {
+        guard !seeAlsoInTopicsHeadings.isEmpty else {
+            return []
+        }
+        return seeAlsoInTopicsHeadings.compactMap { heading -> Problem? in
+            guard let headingRange = heading.range else { return nil }
+            let diagnostic = Diagnostic(
+                source: sourceFile,
+                severity: .warning,
+                range: headingRange,
+                identifier: "org.swift.docc.SeeAlsoInTopicsHeadings",
+                summary: #"Level-3 heading "See Also" can't form a See Also section. Did you mean to use a level-2 heading?"#,
+                explanation: #"A level-2 heading with the name "See Also" is a reserved heading name you use to begin a section to groups related symbols or links. To resolve this issue, change the heading level to a level-2 heading to form a See Also section or change the name of this heading to form a task group."#
+            )
+            let solutions = [
+                Solution(
+                    summary: "Change to level-2 heading",
+                    replacements: [Replacement(range: headingRange, replacement: "## \(heading.plainText)")]
```

Comment:
Perhaps that's ok though, since you'd then get a warning that a 3rd-level heading isn't allowed inside a See Also section. It introduces a bit more unnecessary friction though.

Reply:
Thanks Jack. That’s a good point.
I agree that suggesting a solution to change the heading level would be okay since there would be another warning for the issue that applying the solution introduces.

Suggesting to move the section would get more complicated.

An alternative could be to detect when there are more level-three headings after the “See Also” heading and not offer any solution, but I’m not sure that that provides a better user experience.

Reply:
👍 I agree with ya'll, the majority of the time this implementation will work fine. Replacements across many lines can be dangerous if there's a bug in the range gathering, so the risk is likely not worth the potential reward. We can always change our minds later if we find it's more common than expected.